app/src/main/java/com/moviefiend/torian/moviefiend/MovieDetailsFragment.java

```diff
@@ -1,61 +1,63 @@
 package com.moviefiend.torian.moviefiend;
 
-import android.app.Fragment;
-import android.graphics.Movie;
 import android.os.Bundle;
 import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.ImageView;
-import android.widget.TextView;
 
-import com.moviefiend.torian.moviefiend.network.NowPlayingResponse;
-import com.squareup.picasso.Picasso;
+import com.moviefiend.torian.moviefiend.network.MoviesResponse;
 
-public class MovieDetailsFragment extends Fragment {
+public class MovieDetailsFragment extends Fragment implements MovieDetailsCustomView.Listener {
 
-    public static final int MOVIE_POSTER_SIZE = 780;
+    public interface Listener {
+        void onSimilarMoviesClicked(MoviesResponse.MovieInfo movieInfo);
+    }
+
+    public static final String MOVIE_INFO_ARG = "movie_info";
 
-    private NowPlayingResponse.MovieInfo mMovieInfo;
-    private ImageView mPosterView;
-    private TextView mRatingView;
-    private TextView mDescriptionView;
+    private MoviesResponse.MovieInfo mMovieInfo;
+    private Listener mListener;
 
     @Nullable
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.movie_details_fragment, container, false);
+        MovieDetailsCustomView customView = (MovieDetailsCustomView) view;
 
-        mMovieInfo = getArguments().getParcelable(MovieDetailsActivity.MOVIE_INFO_EXTRA);
-
-        mPosterView = (ImageView) view.findViewById(R.id.poster);
-        mRatingView = (TextView) view.findViewById(R.id.rating);
-        mDescriptionView = (TextView) view.findViewById(R.id.description);
+        mMovieInfo = getArguments().getParcelable(MOVIE_INFO_ARG);
+        customView.setMovie(mMovieInfo);
+        customView.setListener(this);
 
-        setTitle(mMovieInfo.getTitle());
-        setPosterView(UrlHelper.getPosterUrl(MOVIE_POSTER_SIZE, mMovieInfo.getPosterPath()));
-        setRating(mMovieInfo.getRating());
-        setDescription(mMovieInfo.getDescription());
+        if (getUserVisibleHint()) {
```

Comment:
@ifeins There will be no problem if its not part of a `ViewPager`; the visibility hint is not only for that (this fragment is actually used not in a `ViewPager` in the normal details activity).
I do agree about the activity's responsibility (+ its less re-useable as a small part of something) so I'm going to change.

P.S you should always be aware of page selection listeners 😄  they don't fire an event for the first page by default